aiocache/backends/valkey.py

```diff
@@ -0,0 +1,250 @@
+import logging
+import sys
+import time
+from typing import Any, Callable, List, Optional, TYPE_CHECKING
+
+from glide import (
+    ConditionalChange,
+    ExpirySet,
+    ExpiryType,
+    GlideClient,
+    GlideClientConfiguration,
+    Script,
+    Transaction,
+)
+from glide.exceptions import RequestError as IncrbyException
+
+from aiocache.base import API, BaseCache
+from aiocache.serializers import JsonSerializer
+
+if TYPE_CHECKING:  # pragma: no cover
+    from aiocache.serializers import BaseSerializer
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing import Any as Self
+
+
+logger = logging.getLogger(__name__)
+
+
+class ValkeyBackend(BaseCache[str]):
+    def __init__(self, config: GlideClientConfiguration = None, **kwargs):
```

Comment:
I still can't really imagine any reason for passing None here. I'm inclined to just make this required and simplify the rest.